spatial/kdtree/kdtree_user_type_example_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kdtree_test
+
+import (
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/spatial/kdtree"
+)
+
+func Example_accessiblePublicTransport() {
+	// Construct a k-d tree of train station locations
+	// to identify accessible public transport for the
+	// elderly.
+	t := kdtree.New(stations, false)
+
+	// Residence.
+	q := place{lat: 51.501476, lon: -0.140634}
+
+	var keep kdtree.Keeper
+
+	// Find all stations within 0.75 of the residence.
+	keep = kdtree.NewDistKeeper(0.75 * 0.75) // Distances are squared.
+	t.NearestSet(keep, q)
+
+	fmt.Println(`Stations within 750 m of 51.501476N 0.140634W.`)
+	for _, c := range keep.(*kdtree.DistKeeper).Heap {
```

Comment:
Could the type assertion be avoided somehow, e.g. by extending the `Keeper` interface?

Reply:
There is no intrinsic reason that a `Keeper` should have a heap.

Reply:
I meant rather whether `Keeper` should have a method that returns the collected points but now I see that the assertion is done here only because you reuse `keep` for both `DistKeeper` and `NKeeper`.